emcc.py

```diff
@@ -1234,6 +1234,15 @@ def check(input_file):
       if shared.Settings.USE_PTHREADS:
         shared.Settings.FETCH_WORKER_FILE = unsuffixed(os.path.basename(target)) + '.fetch.js'
 
+    if shared.Settings.FETCH:
+      # In asm.js we can use a fetch worker, which is created from the main asm.js
+      # code. That lets us do sync operations by blocking on the worker etc.
+      # In the wasm backend we don't have a fetch worker implemented yet, however,
+      # we can still do basic synchronous fetches in the same places: if we can
+      # block on another thread then we aren't the main thread, and if we aren't
+      # the main thread then synchronous xhrs are legitimate.
+      shared.Settings.USE_FETCH_WORKER = int(shared.Settings.USE_PTHREADS and not shared.Settings.WASM_BACKEND)
```

Comment:
What do you think about making this an overridable option? I like the idea of USE_PTHREADS + NO_WASM_BACKEND without a fetch worker.

Reply:
Interesting, yeah, that makes sense. Changed now to set this to 0 automatically when not needed (like with the wasm backend), but not touch it otherwise, which allows fastcomp users to disable the fetch worker too.